### ISSUE TITLE
Unable to load CryptoLibrary .so file

### ISSUE THREAD

**Issue by Srtake** [s3team/Abacus#6]
Hello,

I try to reproduce the ICSE2021 results following the readme.pdf tutorial, but get errors like this:

`E: Unable to load /abacus/Pintools/CryptoLibrary/obj-ia32/chacha20-monocypher-3.0.so.  Check the architecture type.`

All algorithms except AES cannot be evaluated. Before I got this error, everything seems to be ok. It seems like there is something wrong when building 32-bit executable in my x64 machine. 

I run the code in the supplied Docker container (except the OS is Ubuntu 20.04) in a Ubuntu 20.04 machine. When creating the container, the docker.sh didn't work (it complains that it is unable to locate the package), so I manually run the command in Dockerfile.

Is there any idea to solve this? Thanks.

**Comment by qinkunbao** [comment]
@Srtake Thank you for the question.

Could you run the command?
``
file /abacus/Pintools/CryptoLibrary/obj-ia32/chacha20-monocypher-3.0.so
``
Make sure the so pintool is compiled with 32-bit. Let me know if you still have a question.

**Comment by Srtake** [comment]
Thanks for the reply!
I solved the docker.sh issue (by clearing the cache) and use the docker.sh script to create the Docker container (which is exactly the same as what the `docs/result.md` says). This time I can successfully run AES, DES and RSA algorithms (in all library versions), but the remaining still fail:
- For ARGON2i, CHACHA20, ECDSA, POLY1305 I got the same error as before: `E: Unable to load /abacus/Pintools/CryptoLibrary/obj-ia32/chacha20-monocypher-3.0.so.  Check the architecture type.` 
- I tried to run the `file`command you said and got this result: `/abacus/Pintools/CryptoLibrary/obj-ia32/chacha20-monocypher-3.0.so: ELF 32-bit LSB shared object, Intel 80386, version 1 (SYSV), dynamically linked, BuildID[sha1]=bbf022f49be0a9a1ca09db5e8387ff51d2b9f7b4, not stripped`. It seems like the pintool is compiled with 32bit correctly, and when run the same command with DES pintools (which execute successfully) I got the same result.
- For Ed25519 I got a segmentation fault error: `./start.sh: line 33:    42 Segmentation fault      (core dumped) $qif /abacus/script/$dir_str/Inst_data.txt -o ${dir_str,,}.txt`. Maybe is there a memory issue?
